version: '3'

services:
  server:
    build:
      context: .
      dockerfile: Dockerfile
    restart: on-failure  # перезапуск при ошибках
    depends_on:
      - db
    expose:
      - "${METRICS_PORT}"
    environment:
      KAFKA_BROKERS: "kafka:9092"  
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_HOST: db
      POSTGRES_PORT: "5432"
    networks:
      - kafka_net
      - observability_net


  db:
    image: postgres:16-alpine
    restart: always
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    volumes:
      - ./database:/var/lib/postgresql/data  
    ports:
      - "5432:5432"


networks:
  observability_net:
    external: true
  kafka_net:
    external: true
